### PR TITLE
FieldOverrides UI

### LINKED ISSUE #21111: New Panel Editor Epic
### Todos (in priority order):
- [x] Panel General settings @dprokop https://github.com/grafana/grafana/pull/22870
  - [x] Add possibility to edit panel title *(General panel settings)*
  - [x] Add possibility to edit panel description *(General panel settings)*
  - [x] Add possibility to edit panel links *(General panel settings)*
 - [x]  Add possibility to edit panel repeat options @torkelo (https://github.com/grafana/grafana/pull/22984)
- [x] Show dashboard variables in the panel edit 
- [x] Experiment with displaying option categories (i.e. **Category** > Option name` as VSCode does) https://github.com/grafana/grafana/pull/23145 @dprokop 
- [x] Scrollbar and search in the visualisation tab (#23253) @dprokop 
- [x] Move visualisations list to the sidebar https://github.com/grafana/grafana/pull/23696 @dprokop 
- [x] Core panels docs update review and effort assessement  https://github.com/grafana/grafana/issues/23549 @dprokop 
- [x] Enable Panel Inspector in the new edit mode @torkelo 
- [x] Adjust actions in the Panel Menu https://github.com/grafana/grafana/pull/23888 @dprokop
- [x] Migrate FieldDisplay options component to new Form styles
- [x] Enable overrides remove
- [x] New design for value mappings component
- [x] **!!! Remove NewPanelEditorContext when we switch to new editor** https://github.com/grafana/grafana/pull/23652 @dprokop
- [x] Reuse query results (do no issue new query) when going into edit mode https://github.com/grafana/grafana/issues/21917
- [x] Angular panels are blank after going into edit mode
- [x] Tabs view below visualization for Queries, Alerting, Transformations
- [x] Visualization selection
- [x] Data links list item design with trash can and some minor details https://github.com/grafana/grafana/pull/22304
- [x] Data links, show modal after adding new link https://github.com/grafana/grafana/pull/22304
- [x] Discard / Save mechanic 
- [x] Resize Panes
- [x] Hide / Show side pane
- [x] Shared panel query (@dprokop, 23.04.2020 -> no longer a bug)
- [x] E2E Tests https://github.com/grafana/grafana/pull/23849 @hugohaggmark 

### Bugs:
- [x] Saving dashboard from edit mode works but no changes are saved (as they are not committed yet)  https://github.com/grafana/grafana/issues/23910
- [x] When switching between panels we do not re-trigger "data processing" (ie the function inside getData in PanelQueryRunner). So the new panel's field config options are not applied.  https://github.com/grafana/grafana/pull/23785
- [x] Alerting state & annotations do not show as panel id is different, so the concept of having a different panel id for the model under editing is not going to work
- [x] Test Alert Rule does not work, it pushes the current dashboard json without the current panel edits (and with a mismatch in panel id as the edit panel has different id)
- [ ] When entering edit mode for a panel with annotations or alert the annotations or alert state will not load in editor until you refresh (trigger query). A bit tricky to fix. @hugohaggmark 
- [x] When switching between react -> angular -> other react panel will reset all field configs  
- [x] When switching to React Table panel the thresholds change to percentage mode
- [x] Go to a dashboard open new edit mode on react panel (any), then go another dashboard with a react table. open in new edit mode. Then panel editor crashes. Super strange the PanelEditor gets mounted and unmounted right away and causing an error as the unmount happens before the redux mount ini action is completed.  Need to figure out why it gets unmounted right away. 
- [x] Uncontrolled input error
image
- [x] Clicking on the splitter i makes the visualization disappear 
- [x] Editing an old table panel, fixed
- [x] Query is issued when going to panel edit https://github.com/grafana/grafana/pull/23837 @dprokop 
- [x] Panels with skipDataQuery setting break editor https://github.com/grafana/grafana/pull/23836 @dprokop

### Documentation
- [x] Field overrides docs
- [x] Panel editor docs update
- [x] Core panels docs update
Related issue: https://github.com/grafana/grafana/issues/22269

### Post 7.0 backlog
- Implement search in the panel options sidebar https://github.com/grafana/grafana/issues/25426
- Enable local storage persistence for panel options collapsible sections and query/transformation row UI state https://github.com/grafana/grafana/issues/25427
- Enable queries reorder by drag and drop
- Implement change indicators for options https://github.com/grafana/grafana/issues/25430
- Add categorisation/tags to panels list
- In the Panel Editor the description input and field description is taking up a lot of vertical space, would be nice to have "+ Add description" as a small text link, that when clicked shows the description field label, description & input. (Might be a stupid idea). 
- Add `collapse/expand all` for sections in the sidebar. @sarlinska https://github.com/grafana/grafana/issues/25431
- Problem with sorting visualisations – what if my selected viz type is all the way down in the list? Then it’s hard for me to tell what type my panel is ([feedback from UX community call](https://docs.google.com/document/d/1Qq19Yr0tK0LNcm5cKpQnuMmZovXnjGgtuA1oze9e23s/edit#))
- Adding an option to name overrides (feedback from UX community call) https://github.com/grafana/grafana/issues/25433
- Adding `Zoom in` next to `Zoom out` icon
- Adding double click on vis type to close section @sarlinska https://github.com/grafana/grafana/issues/25434
- ~Dropdowns in `sidebar` should have a width corresponding with it's content.~
- Z`Field tab` in the sidebar should have inputs and dropdowns width corresponding with their content.~

### Feedback from slack
1. The `Staircase` property of a graph should be hidden too if `Lines` is disabled, it stays visable but is not clickable.
Kapture 2020-05-13 at 16 42 02

### PR DESCRIPTION
**Pull request by dprokop**
Added some styling for overrides ui. Maybe not something we wanna stick with but anything's better than red and blue boxes ;) Also fixed the height issue in panel editor sidebar that made the selects being cut

### CODE REVIEW

**Comment by torkelo** [comment]
Wow, super great start! looks amazing! Super great to demo for UX team. We should schedule a call with UX later next week to get them up to speed with what we have done, how they can try it out and what's the things to explore are.

**Comment by dprokop** [comment]
yes, that's my plan!

**Review by torkelo — APPROVED** [review]
😍

File: public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx [hunk 1]
```diff
@@ -265,7 +265,11 @@ export class PanelEditorUnconnected extends PureComponent<Props> {
             >
               {this.renderHorizontalSplit(styles)}
               <div className={styles.panelOptionsPane}>
-                <CustomScrollbar>
+                <CustomScrollbar
+                  className={css`
+                    height: 100% !important;
```
[the remark this answers is not in the document]
Comment (dprokop) [reply public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx]: Yes it should, but it doesn't :/ need to come up with some other solution for this.

[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/PanelEditor.tsx]: But we do not have this in any other place we use custom scrollbar?

**Pull request merged**